### PR TITLE
adding nosqlbench test scripts

### PR DESCRIPTION
**Pull request by jeffreyscarpenter**

### CODE REVIEW

File: nosqlbench/http-jsonapi-crud-basic.yaml
Comment (tatu-at-datastax) [reply nosqlbench/http-jsonapi-crud-basic.yaml:45]: One possible problem is if there is no check that reads should target documents actually written: if not, we'd get 404 (correctly) when trying to access non-existing documents. So return value depends on true existence of document(s). I agree it would be very useful to have sanity checks so that we do not accidentally "test" performance of broken tests (nothing found, ever, which can be fast).

File: nosqlbench/http-jsonapi-crud-dataset.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-crud-dataset.yaml:81]: actually I'm just removing the TODO. Didn't realize ID was optional.

Comment (jshook) [reply nosqlbench/http-jsonapi-crud-dataset.yaml:81]: We can build in some available assertions to handle this, and this is on the radar for our upcoming Accord testing prep.

File: nosqlbench/http-jsonapi-crud-basic.yaml
Comment (ivansenic) [reply nosqlbench/http-jsonapi-crud-basic.yaml:45]: I am not saying specifically for the schema change responses, I am saying lets add it everywhere. But question is actually good, what do we want from these tests?

File: nosqlbench/http-jsonapi-keyvalue.yaml
Comment (ivansenic) [reply nosqlbench/http-jsonapi-keyvalue.yaml]: Guess this is done, resolving.

**Review by ivansenic — APPROVED** [review]
I am approving to unblock, as I will not be here tomorrow. I resolved all the conversations that were addressed. Few unresolved are still there, please resolve before merging.

File: nosqlbench/http-jsonapi-crud-dataset.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-crud-dataset.yaml]: The service never returns a 4xx anyway, according to current spec. It's 200 or 500.

File: nosqlbench/http-jsonapi-crud-basic.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-crud-basic.yaml:45]: As per Shooky's example, we do want to allow empty reads. These are performance tests, not correctness tests. 

By default the tests do exit on error condition, but we can override this with the setting `errors=count` to get statistics on errors which we can then analyze. 

According to the JSON API spec this service is going to return 200 for basically everything except 500 server error. So identifying error conditions per call will require more work. We'll probably need to do something like

```
ok-status: "200"
ok-body: "some regex"
```

IMO that's beyond the scope of this initial PR. I'd just like to get these checked in initially and begin running them regularly, and then we can improve the tests as we go.

**Pull request merged**
